tests/monitoring/monitoring.go

```diff
@@ -638,29 +637,85 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, decorators
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{
+				Patches: []v1.CustomizeComponentsPatch{
+					{
+						ResourceName: virtHandler.deploymentName,
+						ResourceType: "DaemonSet",
+						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
+						Type:         v1.StrategicMergePatchType,
+					},
+				},
+			}
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() string {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vh.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]
+			}, 90*time.Second, 5*time.Second).Should(Equal("does-not-exist"))
+
+			return kv
+		}
+
+		setEmulation := func(status bool) *v1.KubeVirt {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.Configuration.DeveloperConfiguration.UseEmulation = status
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return kv.Spec.Configuration.DeveloperConfiguration.UseEmulation
+			}, 90*time.Second, 5*time.Second).Should(Equal(status))
+
+			return kv
+		}
+
+		restoreVirtHandler := func(kv *v1.KubeVirt) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{}
+			kv, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
 		It("KubeVirtNoAvailableNodesToRunVMs should be triggered when there are no available nodes in the cluster to run VMs.", func() {
+			By("Scaling down virt-handler")
+			kv := disableVirtHandler()
+
 			By("Removing the kubevirt.io/schedulable label from all nodes")
 			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
 			Expect(err).ToNot(HaveOccurred())
-
 			removeSchedulableLabelFromNodes(virtClient, nodes)
 
-			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert exists")
-			verifyAlertExist("KubeVirtNoAvailableNodesToRunVMs")
+			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert exists if emulation is disabled")
+			setEmulation(false)
+			verifyAlertExistWithCustomTime("KubeVirtNoAvailableNodesToRunVMs", 10*time.Minute)
+
+			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert does not exist if emulation is enabled")
```

Comment:
I am not sure this makes sense to me. If there is no virt-handler (or in other words no node schedulable) , but emulation is enabled, it still means we can't run VMs.

Reply:
but that should be caught by virt-handler related alerts

here we are directing attention to the nodes: `KubeVirtNoAvailableNodesToRunVMs`

we want to alert users when none of the nodes in the cluster has the capability of running a VM, but KubeVirt is working in perfect conditions

Reply:
Maybe I am reading the code wrong. Here I see that we disabled virt-handler and removed `schedulable` from the nodes. How does emulation affect this?

Reply:
we disable virt-handler to be able to remove the schedulable label and pretend no hardware virtualization is found on a node

then we enable emulation to be able to run VMs not though hardware virtualization but through emulation

Reply:
```
// This label declares whether a particular node is available for
// scheduling virtual machine instances on it. Used on Node.
NodeSchedulable string = "kubevirt.io/schedulable"
```

This means any Virtual Machine. We don't difference between hardware virtualization and emulation. Meaning in order to reflect `KubeVirtNoAvailableNodesToRunVMs` correctly you need at least one node with the label + (kvm or emulation).

What you are doing in the test is:
1. Remove all virt-handlers
2. Enable emulation
3. Expect alert

This is wrong as we can't run any VMs without virt-handler.

Reply:
so we do not need to check for emulation?
is it enough to check for the NodeSchedulable label in the node to know if that know can have a VM?

p.e.:
hardware virtualization -> has label
no hardware virtualization but emulation enabled -> has label
no hardware virtualization and no emulation -> does not have label
?

Reply:
> so we do not need to check for emulation? is it enough to check for the NodeSchedulable label in the node to know if that know can have a VM?
> 
> p.e.: hardware virtualization -> has label no hardware virtualization but emulation enabled -> has label no hardware virtualization and no emulation -> does not have label ?

This is incorrect. The label is not related to hw virtualization nor emulation. See https://github.com/kubevirt/kubevirt/blob/release-0.59/pkg/virt-handler/heartbeat/heartbeat.go#L126. The label only indicates if virt-handler is present or, if it is present, it indicates readiness.

1. No label at all nodes -> No VM can run (at least a new one)
2.  (label present)
     - KVM device (emulation doesn't affect this case)-> HW virtualization should work 
     - No KVM device but emulation -> VMs can run.
     - No KVM and no emulation -> VMs cannot run.

Reply:
okay
I updated the alert, but in the tests, I'm still removing the virt-handlers so I can freely edit the KVM labels as needed, or else they are overwritten